app/scripts/controllers/bridge-status/bridge-status-controller.ts

```diff
@@ -0,0 +1,316 @@
+    this.messagingSystem.registerActionHandler(
+      `${BRIDGE_STATUS_CONTROLLER_NAME}:startPollingForBridgeTxStatus`,
+      this.startPollingForBridgeTxStatus.bind(this),
+    );
+    this.messagingSystem.registerActionHandler(
+      `${BRIDGE_STATUS_CONTROLLER_NAME}:wipeBridgeStatus`,
+      this.wipeBridgeStatus.bind(this),
+    );
+
+    // Set interval
+    this.setIntervalLength(REFRESH_INTERVAL_MS);
+
+    // If you close the extension, but keep the browser open, the polling continues
+    // If you close the browser, the polling stops
+    // Check for historyItems that do not have a status of complete and restart polling
+    this.#restartPollingForIncompleteHistoryItems();
+  }
+
+  resetState = () => {
+    this.update((_state) => {
+      _state.bridgeStatusState = {
+        ...DEFAULT_BRIDGE_STATUS_CONTROLLER_STATE,
+      };
+    });
+  };
+
+  wipeBridgeStatus = ({
+    address,
+    ignoreNetwork,
+  }: {
+    address: string;
+    ignoreNetwork: boolean;
+  }) => {
+    // Wipe all networks for this address
+    if (ignoreNetwork) {
+      this.update((_state) => {
+        _state.bridgeStatusState = {
+          ...DEFAULT_BRIDGE_STATUS_CONTROLLER_STATE,
+        };
+      });
+    } else {
+      const { selectedNetworkClientId } = this.messagingSystem.call(
+        'NetworkController:getState',
+      );
+      const selectedNetworkClient = this.messagingSystem.call(
+        'NetworkController:getNetworkClientById',
+        selectedNetworkClientId,
+      );
+      const selectedChainId = selectedNetworkClient.configuration.chainId;
+
+      this.#wipeBridgeStatusByChainId(address, selectedChainId);
+    }
+  };
+
+  #restartPollingForIncompleteHistoryItems = () => {
+    // Check for historyItems that do not have a status of complete and restart polling
+    const { bridgeStatusState } = this.state;
+    const historyItems = Object.values(bridgeStatusState.txHistory);
+    const incompleteHistoryItems = historyItems
+      .filter(
+        (historyItem) => historyItem.status.status !== StatusTypes.COMPLETE,
+      )
+      .filter((historyItem) => {
+        // Check if we are already polling this tx, if so, skip restarting polling for that
+        const srcTxHash = historyItem.status.srcChain.txHash;
+        const pollingToken = this.#pollingTokensBySrcTxHash[srcTxHash];
+        return !pollingToken;
+      });
+
+    incompleteHistoryItems.forEach((historyItem) => {
+      const statusRequest = {
+        bridgeId: historyItem.quote.bridgeId,
+        srcTxHash: historyItem.status.srcChain.txHash,
+        bridge: historyItem.quote.bridges[0],
+        srcChainId: historyItem.quote.srcChainId,
+        destChainId: historyItem.quote.destChainId,
+        quote: historyItem.quote,
+        refuel: Boolean(historyItem.quote.refuel),
+      };
+
+      const hexSourceChainId = new Numeric(statusRequest.srcChainId, 10)
+        .toPrefixedHexString()
+        .toLowerCase() as `0x${string}`;
```

Comment:
Is there a shared util for this or can you make one for bridging? We do this conversion frequently in components as well

Reply:
Done in [8fc0d07](https://github.com/MetaMask/metamask-extension/pull/28636/commits/8fc0d07ea4b4d1b5102f8a9fd58f186328b7bf22)